config_examples/spanmetrics.yaml

```diff
@@ -0,0 +1,70 @@
+receivers:
+  otlp:
+    protocols:
+      grpc:
+        endpoint: 0.0.0.0:4317
+      http:
+        endpoint: 0.0.0.0:4318
+
+processors:
+  transform:
+    metric_statements:
+    - context: metric
+      statements:
+      # Extract sum and count metrics from the service. We can use these metrics
+      # to get the average request duration by using a metric query like so:
+      # requests.duration_sum:splitBy("service.name")/requests.duration_count.gauge:splitBy("service.name")
+      # Note that the resulting values in the above query will be unitless.
+      - extract_count_metric(false) where type == METRIC_DATA_TYPE_HISTOGRAM
+      - extract_sum_metric(false) where type == METRIC_DATA_TYPE_HISTOGRAM
+  filter:
+    metrics:
+      metric:
+      # The Dynatrace OTLP metrics ingest doesn't currently support histograms, so filter them out.
+      - type == METRIC_DATA_TYPE_HISTOGRAM
+  transform/spanmetrics:
+    metric_statements:
+    - context: datapoint
+      statements:
+      # Not required, but simplifies queries in Dynatrace.
+      - convert_sum_to_gauge() where IsMatch(metric.name, "^requests.*") and metric.type == METRIC_DATA_TYPE_SUM
```

Comment:
Not sure I understand the reasoning behind this. If gauge is semantically correct that is one thing, but I don't think it is in this case right? Why is the querying easier for gauges than sums? Is this something our users are likely to understand when they see it?

Reply:
To address each of your questions:
1. No, gauges are not semantically correct here; both of these values can be aggregated.
2. I can't answer that, but converting the `_sum` metric from a sum to a gauge does remove the need for a `.count` suffix.
3. I tried to make the comment as detailed as possible, but I can't say that I fully understand this, so I wouldn't expect our users to understand it either.